Userland/Libraries/LibJS/Runtime/AtomicsObject.cpp

```diff
@@ -401,6 +461,75 @@ JS_DEFINE_NATIVE_FUNCTION(AtomicsObject::sub)
     VERIFY_NOT_REACHED();
 }
 
+// 25.4.13 Atomics.wait ( typedArray, index, value, timeout ), https://tc39.es/ecma262/#sec-atomics.wait
+JS_DEFINE_NATIVE_FUNCTION(AtomicsObject::wait)
+{
+    auto* typed_array = TRY(typed_array_from(vm, vm.argument(0)));
+    auto index = vm.argument(1);
+    auto value = vm.argument(2);
+    auto timeout = vm.argument(3);
+
+    // 1. Return ? DoWait(sync, typedArray, index, value, timeout).
+    return TRY(do_wait(vm, WaitMode::Sync, *typed_array, index, value, timeout));
+}
+
+// 25.4.14 Atomics.waitAsync ( typedArray, index, value, timeout ), https://tc39.es/ecma262/#sec-atomics.waitasync
+JS_DEFINE_NATIVE_FUNCTION(AtomicsObject::wait_async)
+{
+    auto* typed_array = TRY(typed_array_from(vm, vm.argument(0)));
+    auto index = vm.argument(1);
+    auto value = vm.argument(2);
+    auto timeout = vm.argument(3);
+
+    // 1. Return ? DoWait(async, typedArray, index, value, timeout).
+    return TRY(do_wait(vm, WaitMode::Async, *typed_array, index, value, timeout));
+}
+
+// 25.4.15 Atomics.notify ( typedArray, index, count ), https://tc39.es/ecma262/#sec-atomics.notify
+JS_DEFINE_NATIVE_FUNCTION(AtomicsObject::notify)
+{
+    auto* typed_array = TRY(typed_array_from(vm, vm.argument(0)));
+    auto index = vm.argument(1);
+    auto count_value = vm.argument(2);
+
+    // 1. Let byteIndexInBuffer be ? ValidateAtomicAccessOnIntegerTypedArray(typedArray, index, true).
+    // FIXME: ValidateAtomicAccessOnIntegerTypedArray is a new AO from the resizable array buffer proposal. Use it when the proposal is implemented.
+    TRY(validate_integer_typed_array(vm, *typed_array, true));
+    auto byte_index_in_buffer = TRY(validate_atomic_access(vm, *typed_array, index));
+
+    // 2. If count is undefined, then
+    double count = 0.0;
+    if (count_value.is_undefined()) {
+        // a. Let c be +∞.
+        count = js_infinity().as_double();
+    }
+    // 3. Else,
+    else {
+        // a. Let intCount be ? ToIntegerOrInfinity(count).
+        auto int_count = TRY(count_value.to_integer_or_infinity(vm));
+
+        // b. Let c be max(intCount, 0).
+        count = max(int_count, 0.0);
+    }
+
+    // 4. Let buffer be typedArray.[[ViewedArrayBuffer]].
+    auto* buffer = typed_array->viewed_array_buffer();
+
+    // 5. Let block be buffer.[[ArrayBufferData]].
+    auto& block = buffer->buffer();
+
+    // 6. If IsSharedArrayBuffer(buffer) is false, return +0𝔽.
+    if (!buffer->is_shared_array_buffer())
+        return Value { 0 };
+
+    // FIXME: Implement the remaining steps when we support SharedArrayBuffer.
+    (void)byte_index_in_buffer;
+    (void)count;
+    (void)block;
+
+    return vm.throw_completion<InternalError>(ErrorType::NotImplemented, "SharedArrayBuffer"sv);
+}
+
 // 25.4.14 Atomics.xor ( typedArray, index, value ), https://tc39.es/ecma262/#sec-atomics.xor
```

Comment:
Should this be `25.4.16` now?

Reply:
It should, there's a bunch numbers that we're missing - I have another branch where I've gone through and updated them all, so I left them alone here

Reply:
Fair enough :^)